eng/scripts/initialize.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# This script is used to execute mypy within a tox environment. Depending on which package is being executed against,
+# a failure may be suppressed.
+
+import os
+from pathlib import Path
+from helpers import call, str2bool
+import argparse
+import shutil
+
+root_dir = Path.resolve(Path(__file__) / "../../..")
+
+
+def copy(src: Path, dest: Path) -> None:
+    dest.parent.mkdir(parents=True, exist_ok=True)
+    shutil.copy(src, dest)
+
+
+def main(update_to_latest_typespec: bool, build_artifacts_path: Path) -> None:
+    # Install global dependencies
+    call("npm install -g pnpm@8.3.1")
+    call("npm install -g autorest@3.6.3")
+    call("npm install -g @typespec/compiler@0.49.0")
+
+    if build_artifacts_path:
+        # copy package.json and pnpm-lock.yaml from build artifacts
+        lock_files_dir = Path(build_artifacts_path) / "lock-files"
+        if lock_files_dir.exists():
+            print(f"Copying package.json and pnmp-lock.yaml from {lock_files_dir}")
+            copy(lock_files_dir / "package.json", root_dir)
+            copy(lock_files_dir / "pnpm-lock.yaml", root_dir)
+            copy(
+                lock_files_dir / "emitter/package.json",
+                root_dir / "packages/typespec-python",
+            )
+
+        # Pnpm install
+        call("pnpm install --frozen-lockfile")
+    else:
+        if update_to_latest_typespec:
+            # Update typespec packages to latest dev version
+            call(
+                "npx -y @azure-tools/typespec-bump-deps@0.4.0 --use-peer-ranges package.json packages/typespec-python/package.json"
+            )
+
+            # Pnpm install
+            call("pnpm install --no-frozen-lockfile")
```

Comment:
is there a big reason why we shouldn't just run `pnpm install`, regardless of dependency updates? Does it break anything or is it make sure we don't have a lock file diff when we check the git diff after generation

Reply:
if unspecified pnpm will default `--frozen-lockfile` to `true` in CI and `false` in non-ci.  It determines CI by looking for common CI environment variables: https://pnpm.io/cli/install#--frozen-lockfile

We want to be explicit as this script should be runnable locally and in ci.